pkg/eventprocessor/impl/executor/impl/txnscope.go

```diff
@@ -0,0 +1,109 @@
+package impl
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"github.com/textileio/go-tableland/internal/tableland"
+	"github.com/textileio/go-tableland/pkg/eventprocessor/eventfeed"
+	"github.com/textileio/go-tableland/pkg/eventprocessor/impl/executor"
+	"github.com/textileio/go-tableland/pkg/parsing"
+	"github.com/textileio/go-tableland/pkg/tables/impl/ethereum"
+)
+
+var tableIDIsEmpty = "table id is empty"
+
+// errQueryExecution is an error returned when the query execution failed
+// with a cause related to th query itself. Retrying the execution of this query
+// will always return an error (e.g: inserting a string in an integer column).
+// A query execution failure due to the database being down or any other infrastructure
+// problem isn't an ErrQueryExecution error.
+type errQueryExecution struct {
+	Code string
+	Msg  string
+}
+
+// Error returns a string representation of the query execution error.
+func (e *errQueryExecution) Error() string {
+	return fmt.Sprintf("query execution failed with code %s: %s", e.Code, e.Msg)
+}
+
+type txnScope struct {
+	log zerolog.Logger
+
+	parser    parsing.SQLValidator
+	acl       tableland.ACL
+	scopeVars scopeVars
+
+	txn *sql.Tx
+}
+
+type eventExecutionResult struct {
+	TableID *tableland.TableID
+	Error   *string
+}
+
+func (ts *txnScope) executeTxnEvents(
```

Comment:
Now the event type detection and routing to execution logic happens at this layer, since it's a responsibility at the EVM transaction scope.